pkg/operator/starter_controller.go

```diff
@@ -0,0 +1,121 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/csi-operator/pkg/operator/config"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/csi/csicontrollerset"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+)
+
+type Controller struct {
+	operatorClient           v1helpers.OperatorClient
+	csiControllers           []*csicontrollerset.CSIControllerSet
+	factoryControllers       []factory.Controller
+	controllersRunning       bool
+	eventRecorder            events.Recorder
+	operatorControllerConfig *config.OperatorControllerConfig
+}
+
+type Runnable interface {
+	Run(ctx context.Context, workers int)
+}
+
+const (
+	// Minimal interval between controller resyncs.
+	resyncInterval = 20 * time.Minute
+
+	operatorConditionPrefix = "CSIController"
+)
+
+func StarterController(operatorClient v1helpers.OperatorClient,
```

Comment:
Could we get a doc string explaining what this is doing? fwiw, it is checking that the preconditions, if any, are met and then starting/stopping every other controller accordingly?

Reply:
Yes, that's the point of this controller